src/server/game/Entities/Player/Player.cpp

```diff
@@ -6379,8 +6379,8 @@ void Player::CheckAreaExploreAndOutdoor()
     if (IsInFlight())
         return;
 
-    bool isOutdoor;
-    uint32 areaId = GetBaseMap()->GetAreaId(GetPositionX(), GetPositionY(), GetPositionZ(), &isOutdoor);
```

Comment:
Hrm. What is the call stack on this? It being included in `GetAreaId` implies that we get this from the vmap lookup... so we can just fetch it in the liquid state update, which happens on relocation for all units anyway.

I will investigate.

Reply:
This whole function is called in Player::UpdatePosition
Which checks for m_needsZoneUpdate and updates the zone if set to true (I did not check if that implementation is correct tho)